src/core/server/saved_objects/migrationsv2/model/extract_errors.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { TransformErrorObjects } from '../../migrations/core';
+import { CheckForUnknownDocsResponseDoc } from '../actions';
+
+/**
+ * Constructs migration failure message strings from corrupt document ids and document transformation errors
+ */
+export function extractTransformFailuresReason(
+  corruptDocumentIds: string[],
+  transformErrors: TransformErrorObjects[]
+): string {
+  const corruptDocumentIdReason =
+    corruptDocumentIds.length > 0
+      ? ` ${
+          corruptDocumentIds.length
+        } corrupt saved object documents were found: ${corruptDocumentIds.join(',')}`
+      : '';
+  // we have both the saved object Id and the stack trace in each `transformErrors` item.
+  const transformErrorsReason =
+    transformErrors.length > 0
+      ? ` ${transformErrors.length} transformation errors were encountered:\n ` +
+        transformErrors
+          .map((errObj) => `- ${errObj.rawId}: ${errObj.err.stack ?? errObj.err.message}\n`)
+          .join('')
+      : '';
+  return (
+    `Migrations failed. Reason:${corruptDocumentIdReason}${transformErrorsReason}\n` +
+    `To allow migrations to proceed, please delete or fix these documents.`
+  );
+}
+
+export function extractUnknownDocFailureReason(
+  unknownDocs: CheckForUnknownDocsResponseDoc[],
+  sourceIndex: string
+): string {
+  return (
+    `Migration failed because documents from unknown types were found. ` +
```

Comment:
I think we can be clearer with the failure message here. e.g. 
```suggestion
    `Migration failed because documents were found for unknown saved object types. ` +
```

Reply:
If we want to be really verbose, we could hint at the cause too:
```
The plugin or feature that was used to create these documents is no longer available.
```
That might be taking it a bit far though. WDYT?

Reply:
> The plugin or feature that was used to create these documents is no longer available.

We can't really know for sure why and how the type is no longer registered, which is why I would ideally avoid taking such guess in the outputted logs

e.g
1. The registering plugin would just be disabled
2. The document would be corrupted and just not have any `type` at all
3. ???

Reply:
It could probably be summarized as: `the plugin that created this document is either missing or disabled.` Any other case is likely a bug (either in our plugin code or a 3rd party plugin). We need to have an explicit API for registering types that can be safely deleted.